### ISSUE TITLE
CoAtNet not present in `timm`

### ISSUE THREAD

**Issue by sayakpaul** [sayannath/CoAtNet-GSoC#5]
@gustheman @sayannath 

It turns out that `timm` doesn't have any implementation of CoAtNet which means we really don't have any pre-trained weights available for the model at the moment. It was an oversight on my end to confuse [this model](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/coat.py) with CoAtNet. 

We can do either of the following now:

* Continue with the CoAtNet implementation and train it on smaller datasets to confirm it works. I mention smaller datasets because that way we can complete things within the timeline which doesn't factor in ImageNet-1k related tasks at the moment.
* Decide on another image classification model and follow the same plan as CoAtNet.

Let's discuss.

**Comment by sayannath** [comment]
Cool! On it.

**Comment by sayakpaul** [comment]
What are you on? Didn't get you.

**Comment by sayannath** [comment]
On implementing the model asap.

**Comment by sayakpaul** [comment]
No, I think it's better to wait for Gus's comments here. Since the pre-trained model parameters aren't available we need to take an informed decision.

**Comment by gustheman** [comment]
I'm fine with changing to another model to make this achievable!
Any suggestions of other possible ones not on TF Hub yet?

**Comment by sayannath** [comment]
MobileViT can be an option.

**Comment by sayakpaul** [comment]
Yes sounds good to me. MobileViTs are available in official capacity as well. 

@gustheman WDYT?

**Comment by gustheman** [comment]
I agree, let's go with MobileViT then!
